batcher/aligned-sdk/src/communication/batch.rs

```diff
@@ -0,0 +1,143 @@
+use std::{collections::HashSet, time::Duration};
+
+use futures_util::StreamExt;
+use log::{debug, error};
+use tokio::time::timeout;
+
+use crate::{
+    core::errors,
+    core::types::{
+        AlignedVerificationData, BatchInclusionData, VerificationCommitmentBatch,
+        VerificationDataCommitment,
+    },
+    eth::BatchVerifiedEventStream,
+};
+
+const AWAIT_BATCH_VERIFICATION_TIMEOUT: u64 = 60;
+
+pub async fn handle_batch_inclusion_data<'s>(
+    batch_inclusion_data: BatchInclusionData,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+    stream: &mut BatchVerifiedEventStream<'s>,
+    verified_batch_merkle_roots: &mut HashSet<Vec<u8>>,
+) -> Result<(), errors::SubmitError> {
+    debug!("Received response from batcher");
+    debug!(
+        "Batch merkle root: {}",
+        hex::encode(batch_inclusion_data.batch_merkle_root)
+    );
+    debug!("Index in batch: {}", batch_inclusion_data.index_in_batch);
+
+    let verification_data_commitment = verification_data_commitments_rev.pop().unwrap_or_default();
```

Comment:
Should return an error if empty instead of doing unwrap_or_default

Reply:
Solved in [a95356d](https://github.com/yetanotherco/aligned_layer/pull/583/commits/a95356d1eae11898fb88283ce184123dd07ba6c9).